client/src/components/formHelpers/form-validators.test.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  editorValidator,
+  fCCValidator,
+  localhostValidator,
+  httpValidator,
+  pathValidator
+} from './form-validators';
+
+test('editorValidator', () => {
+  const editorAddresses = [
+    'repl.it/@username',
+    'repl.it/join/username',
+    'replit.com/@username',
+    'replit.com/join/username',
+    'glitch.com/edit/#!',
+    'codesandbox.io/s/',
+    'https://github.com',
+    'https://glitch.com/edit/#!/project',
+    'https://glitch.com/edit/#!'
+  ];
+
+  const nonEditorAddresses = [
+    'example.com',
+    'http://example.com',
+    'https://example.com',
+    'http://repl.it',
+    'http://glitch.com',
+    'https://repl.it/home'
+  ];
+
+  editorAddresses.forEach(address => {
```

Comment:
this is exactly period im learning to do unit tests,
very good to see it in action!
may i ask: what to use `toBeNull` better than`toBeTruthy`?
Thanks in advance

Reply:
> may i ask: what to use toBeNull better thantoBeTruthy?

1) The error (assertion) messages are better - the error explicitly mentions the expectation for the value to be `null`
2) The API does explicitly return `null`
3) Any change to the API where it accidentally returns `undefined` or `false` is now caught by the tests